src/main/java/xyz/nucleoid/plasmid/util/BucketFind.java

```diff
@@ -0,0 +1,43 @@
+package xyz.nucleoid.plasmid.util;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.math.BlockPos;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class BucketFind {
```

Comment:
Since this is a utility class, it should not be extendable or constructable:

```suggestion
public final class BucketFind {
    private BucketFind() {}

```

Reply:
This was only partially fixed in 3145459e790e7740817617dd1e6999ad831ae704; in addition to the `final` modifier on the class, the constructor should be made private as well.